src/domain/core/src/entities/events.rs

```diff
@@ -0,0 +1,19 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use opendatafabric::DatasetID;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone)]
+pub struct DatasetEventRemoved {
```

Comment:
1) The method on repository is called `delete_dataset`, so let's use "Deleted" here for consistency?

2) If there will not be `DatasetEvent` enum, should we just name it `DatasetDeleted`? Perhaps having it in `events` module is enough to classify it as an event and we don't need a special naming scheme/prefix?

Reply:
I've renamed as recommended, but the end result does not make much sense either:

```
    b.bind::<dyn AsyncEventHandler<TaskFinished>, ...;
    b.bind::<dyn AsyncEventHandler<DatasetDeleted>, ...;
    b.bind::<dyn AsyncEventHandler<UpdateScheduleEventModified>, ...;
```

My gut feeling is to have a uniform naming scheme for events, regardless of whether they belong to enum or not.

Reply:
What do you think about **requiring** that events of a certain sub-domain are in `enum`?

While we don't use even sourcing in `DatasetRepository` now - potentially we will soon (e.g. when dataset is renamed we would want old URLs in the platform to redirect to new name until old name is taken). So we will likely have a `DatasetRepositoryEvent` enum.

Also when I look at `AsyncEventHandler<UpdateScheduleEventModified>` - this seems very granular to me. Perhaps event handlers should always operate on enum level: `AsyncEventHandler<UpdateScheduleEvent>`?

While this will dispatch some events a handler might not be interested in, it had a significant added bonus: event handlers will have to `match` on event type and the compiler will force us to consider whether a newly-introduced event needs to be handled by certain logic or not.

Reply:
> What do you think about requiring that events of a certain sub-domain are in enum?

I don't see benefits. If we convert the `DatasetRepository` to ES (hard to imagine the cost of it), then this idea will be relevant.

>> Perhaps event handlers should always operate on enum level: AsyncEventHandler<UpdateScheduleEvent>?

Again, I don't see any benefits. Right now I'm very specific about the type of event I want to get, and it gives me perfect type checking. Introducing a new event in the same aggregate does not imply that old handlers would be interested in handling it as well.

Reply:
I've added an enum for dataset events, and unified naming in all 3 event domains.

Reply:
I understand the match principle, that's very often used to safely match enums, however both strategies for events are valid, and on today I do not see cases of full aggregate handlers, but I clearly see services which pick 1 event from N aggregates.